lib/lita/handlers/jenkins.rb

```diff
@@ -12,27 +12,37 @@ class Jenkins < Handler
         'jenkins list <filter>' => 'lists Jenkins jobs'
       }
 
-      route /j(?:enkins)? b(?:uild)? ([\w\.\-_ ]+)/i, :jenkins_build, command: true, help: {
+      route /j(?:enkins)? b(?:uild)? ([\w\.\-_ ]+)(, (.+))?/i, :jenkins_build, command: true, help: {
         'jenkins b(uild) <job_id or job_name>' => 'builds the job specified by ID or name. List jobs to get ID.'
       }
 
-      def jenkins_build(response)
-        job = find_job(response.matches.last.last)
+      def jenkins_build(response, empty_params = false)
```

Comment:
One benefit is that you can then call it without the optional parameter: `jenkins_build(response)` and it means the same thing as `jenkins_build(response, empty_params: false)`. And when you do want to call it with the parameter explicitly it's more clear what the parameter is: `jenkins_build(response, empty_params: true)`.

But that's not really the reason I was asking to look into it. It's more because the code is a little confusing to me (tho that's really my fault because I'm not very familiar with it anymore).

Another smell is `params` on line 29 and how is can be either a hash or true/false.

But I'm taking too long to respond so let's merge this in and then we can work on refactor as we move forward.

Btw, do we need/have a test for the 400 response case?